KAFKA-9663: Doc some null returns in KafkaStreams

tombentley opened the pull request:
Documents the possibility of null return value in KafkaStreams.getMetadataForKey(), and KafkaStreams.queryMetadataForKey().

This fixes KAFKA-9663.

tombentley commented:
@guozhangwang, @mjsax please could you review?

mjsax commented on streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java:
```diff
@@ -1083,7 +1083,8 @@ public void cleanUp() {
      * @return {@link StreamsMetadata} for the {@code KafkaStreams} instance with the provided {@code storeName} and
-     * {@code key} of this application or {@link StreamsMetadata#NOT_AVAILABLE} if Kafka Streams is (re-)initializing
+     * {@code key} of this application or {@link StreamsMetadata#NOT_AVAILABLE} if Kafka Streams is (re-)initializing,
+     * or null if no matching metadata could be found.
```
Nit: `null` -> `{@code null}`

Same for all other cases

tombentley committed:
KAFKA-9663: Doc some null returns in KafkaStreams

Documents the possibility of null return value in
KafkaStreams.getMetadataForKey(), and KafkaStreams.queryMetadataForKey().

This fixes KAFKA-9663.

tombentley commented:
Closed in error. Now fixed @mjsax.

mjsax commented:
Thanks for the PR @tombentley! Merged to `trunk`.
